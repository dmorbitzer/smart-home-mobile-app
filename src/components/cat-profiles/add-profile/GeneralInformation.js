import { TextInput, Text } from "@react-native-material/core";
import { useState } from "react";
import { View, StyleSheet } from "react-native";
import DropDownPicker from "react-native-dropdown-picker";
import NumericInput from "react-native-numeric-input";
import { HelperText, useTheme } from "react-native-paper";
import { DatePickerInput } from "react-native-paper-dates";

export default function GeneralInformation(props) {
  const theme = useTheme();
  const [open, setOpen] = useState(false);
  const genderItems = props.genders.map((element) => {
    return {
      label:
        element.node.name === "male"
          ? "Männlich"
          : element.node.name === "female"
          ? "Weiblich"
          : "",
      value: element.node.id,
    };
  });

  return (
    <View>
      <Text style={styles.label}>Name:</Text>
      <TextInput
        color={theme.colors.primary}
        inputContainerStyle={{ backgroundColor: theme.colors.background }}
        onChangeText={props.setCatName}
        value={props.catName}
        style={styles.input}
      />
      <HelperText type="error" visible={props.hasCatNameError}>
        Katzennahme fehlt!
      </HelperText>

      <Text style={styles.label}>Geburtsdatum:</Text>
      <View style={styles.input}>
        <DatePickerInput
          locale="de"
          label={null}
          value={props.catBirthdate}
          onChange={props.setCatBirthdate}
          inputMode="start"
          color="red"
          inputEnabled={false}
        />
      </View>

      <HelperText type="error" visible={props.hasBirthdateError}>
        Geburtsdatum fehlt!
      </HelperText>

      <View style={styles.dropdownView}>
        <Text style={styles.label}>Geschlecht:</Text>
        <DropDownPicker
          open={open}
          value={props.catSex}
          items={genderItems}
          setOpen={setOpen}
          setValue={props.setCatSex}
          placeholder={
            props.placeholder ? props.placeholder : "Geschlecht wählen"
          }
          containerStyle={styles.input}
          style={{ backgroundColor: theme.colors.background }}
          listItemContainerStyle={{ backgroundColor: theme.colors.background }}
          listMode="SCROLLVIEW"
        />
        <HelperText type="error" visible={props.hasCatSexError}>
          Katzengeschlecht fehlt!
        </HelperText>
      </View>

      <Text style={styles.label}>Rasse:</Text>
      <TextInput
        color={theme.colors.primary}
        inputContainerStyle={{ backgroundColor: theme.colors.background }}
        onChangeText={props.setCatRace}
        value={props.catRace}
        style={styles.input}
      />
      <HelperText type="error" visible={props.hasCatRaceError}>
        Katzenrasse fehlt!
      </HelperText>

      <Text style={styles.label}>Gewicht in Kilogram:</Text>
      <NumericInput
        value={props.catWeight}
        minValue={0}
        maxValue={10}
        onChange={(value) => props.setCatWeight(value)}
        containerStyle={styles.input}
        valueType="real"
        totalWidth={150}
        totalHeight={50}
        iconSize={25}
        step={0.1}
      />
      <HelperText type="error" visible={props.hasCatWeightError}>
        Katzengewicht in Kilogram fehlt!
      </HelperText>
    </View>
  );
}

const styles = StyleSheet.create({
  input: {
    marginTop: 20,
  },
  label: {
    marginTop: 20,
    fontWeight: "bold",
  },
  dropdownView: {
    zIndex: 20,
  },
});
