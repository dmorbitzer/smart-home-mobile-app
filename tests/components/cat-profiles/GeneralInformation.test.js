import React from "react";
import renderer from "react-test-renderer";

import GeneralInformation from "../../../src/components/cat-profiles/add-profile/GeneralInformation";

jest.useFakeTimers();

const genders = [
  { node: { id: 1, name: "female" } },
  { node: { id: 2, name: "male" } },
];

test("renders CatProfiles correctly", () => {
  const tree = renderer
    .create(<GeneralInformation genders={genders} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
