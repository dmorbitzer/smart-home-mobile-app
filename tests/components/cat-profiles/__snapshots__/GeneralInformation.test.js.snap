// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CatProfiles correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "fontWeight": "bold",
          "marginTop": 20,
        },
      ]
    }
  >
    Name:
  </Text>
  <View
    style={
      [
        {
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#f5f5f5",
            "flexDirection": "row",
          },
          {
            "borderBottomEndRadius": 4,
            "borderBottomStartRadius": 4,
            "borderTopEndRadius": 4,
            "borderTopStartRadius": 4,
          },
          {
            "borderBottomEndRadius": 0,
            "borderBottomStartRadius": 0,
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholderTextColor="#999999"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "color": "#212121",
              "flex": 1,
              "minHeight": 56,
              "paddingEnd": 16,
              "paddingStart": 16,
            },
            {
              "fontSize": 16,
              "letterSpacing": 0.15,
            },
            {
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      />
      <View
        pointerEvents="none"
        style={
          [
            {
              "backgroundColor": "#949494",
              "bottom": 0,
              "end": 0,
              "height": 1,
              "position": "absolute",
              "start": 0,
            },
          ]
        }
      />
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "bottom": 0,
            "end": 0,
            "height": 2,
            "position": "absolute",
            "start": 0,
            "transform": [
              {
                "scaleX": 0,
              },
            ],
          }
        }
      />
    </View>
    <View
      style={
        {
          "marginHorizontal": 16,
          "marginTop": 4,
        }
      }
    />
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Katzennahme fehlt!
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "fontWeight": "bold",
          "marginTop": 20,
        },
      ]
    }
  >
    Geburtsdatum:
  </Text>
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        {
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(28, 27, 31, 0.04)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              {
                "flexGrow": 1,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(28, 27, 31, 0.38)",
                "bottom": 0,
                "height": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": [
                  {
                    "scaleY": 0.5,
                  },
                ],
                "zIndex": 1,
              }
            }
            testID="text-input-underline"
          />
          <View
            style={
              [
                {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                {
                  "minHeight": 56,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  {
                    "color": "rgba(28, 27, 31, 0.38)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 30,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-flat-label-active"
              >
                DD.MM.YYYY
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  {
                    "color": "rgba(28, 27, 31, 0.38)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 30,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-flat-label-inactive"
              >
                DD.MM.YYYY
              </Text>
            </View>
            <TextInput
              color="red"
              editable={false}
              keyboardAppearance="default"
              keyboardType="number-pad"
              maxFontSizeMultiplier={1.5}
              maxLength={10}
              multiline={false}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=" "
              placeholderTextColor="rgba(28, 27, 31, 0.38)"
              selectionColor="rgba(28, 27, 31, 0.38)"
              style={
                [
                  {
                    "margin": 0,
                  },
                  {
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  },
                  {
                    "height": 56,
                  },
                  {
                    "paddingBottom": 4,
                    "paddingTop": 24,
                  },
                  {
                    "color": "rgba(28, 27, 31, 0.38)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  [
                    {},
                  ],
                  undefined,
                ]
              }
              testID="text-input-flat"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "alignSelf": undefined,
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "bottom": undefined,
              "end": undefined,
              "flex": undefined,
              "height": 40,
              "left": undefined,
              "margin": 6,
              "opacity": undefined,
              "position": "absolute",
              "right": 0,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
              "transform": undefined,
              "width": 40,
            }
          }
          testID="icon-button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "flex": 1,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "zIndex": 10,
              }
            }
            testID="icon-button-container"
          >
            <View
              accessibilityComponentType="button"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityTraits="button"
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  false,
                  [
                    {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    {
                      "borderRadius": 20,
                    },
                  ],
                ]
              }
              testID="icon-button"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        presentationStyle="overFullScreen"
        statusBarTranslucent={true}
        supportedOrientations={
          [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      />
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Geburtsdatum fehlt!
  </Text>
  <View
    style={
      {
        "zIndex": 20,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          {
            "color": "#000000",
          },
          {
            "fontWeight": "bold",
            "marginTop": 20,
          },
        ]
      }
    >
      Geschlecht:
    </Text>
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "zIndex": 5000,
          },
          {
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "#000",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 3,
            "width": "100%",
            "zIndex": 5000,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
                "flex": 1,
              },
              {},
              false,
              {},
            ]
          }
        >
          Geschlecht wählen
        </Text>
        <View
          style={
            [
              {
                "marginLeft": 10,
              },
              {},
            ]
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "height": 20,
                  "width": 20,
                },
                {},
              ]
            }
          />
        </View>
      </View>
    </View>
    <Text
      collapsable={false}
      maxFontSizeMultiplier={1.5}
      onLayout={[Function]}
      style={
        {
          "color": "rgba(179, 38, 30, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 4,
          "textAlign": "left",
          "transform": [
            {
              "translateY": -0,
            },
          ],
          "writingDirection": "ltr",
        }
      }
    >
      Katzengeschlecht fehlt!
    </Text>
  </View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "fontWeight": "bold",
          "marginTop": 20,
        },
      ]
    }
  >
    Rasse:
  </Text>
  <View
    style={
      [
        {
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#f5f5f5",
            "flexDirection": "row",
          },
          {
            "borderBottomEndRadius": 4,
            "borderBottomStartRadius": 4,
            "borderTopEndRadius": 4,
            "borderTopStartRadius": 4,
          },
          {
            "borderBottomEndRadius": 0,
            "borderBottomStartRadius": 0,
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholderTextColor="#999999"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "color": "#212121",
              "flex": 1,
              "minHeight": 56,
              "paddingEnd": 16,
              "paddingStart": 16,
            },
            {
              "fontSize": 16,
              "letterSpacing": 0.15,
            },
            {
              "paddingTop": 0,
            },
            undefined,
          ]
        }
      />
      <View
        pointerEvents="none"
        style={
          [
            {
              "backgroundColor": "#949494",
              "bottom": 0,
              "end": 0,
              "height": 1,
              "position": "absolute",
              "start": 0,
            },
          ]
        }
      />
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "bottom": 0,
            "end": 0,
            "height": 2,
            "position": "absolute",
            "start": 0,
            "transform": [
              {
                "scaleX": 0,
              },
            ],
          }
        }
      />
    </View>
    <View
      style={
        {
          "marginHorizontal": 16,
          "marginTop": 4,
        }
      }
    />
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Katzenrasse fehlt!
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "letterSpacing": 0.5,
        },
        {
          "color": "#000000",
        },
        {
          "fontWeight": "bold",
          "marginTop": 20,
        },
      ]
    }
  >
    Gewicht in Kilogram:
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        {
          "borderColor": "#d4d4d4",
          "height": 50,
          "width": 150,
        },
        {},
        {
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderWidth": 0,
          "height": 48,
          "justifyContent": "center",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "width": 45,
          "zIndex": -1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        [
          {
            "alignSelf": "center",
            "borderLeftColor": "#d4d4d4",
            "borderLeftWidth": 1,
            "borderRightColor": "#d4d4d4",
            "borderRightWidth": 1,
          },
        ]
      }
    >
      <TextInput
        editable={true}
        keyboardType="numeric"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        returnKeyType="done"
        style={
          [
            {
              "padding": 0,
              "textAlign": "center",
            },
            {
              "borderLeftColor": "#d4d4d4",
              "borderLeftWidth": 1,
              "borderRightColor": "#d4d4d4",
              "borderRightWidth": 1,
              "color": "black",
              "fontSize": 19,
              "height": 50,
              "width": 60,
            },
            {},
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="0"
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderWidth": 0,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "width": 45,
          "zIndex": -1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      {
        "color": "rgba(179, 38, 30, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.25,
        "lineHeight": 20,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": [
          {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Katzengewicht in Kilogram fehlt!
  </Text>
</View>
`;
